lib/stages/RawReader.cpp

```diff
@@ -343,37 +316,13 @@ void visRawReader::main_thread() {
             // Copy the data from the file
             std::memcpy(frame, mapped_file + file_ind * file_frame_size + metadata_size + 1,
                         data_size);
-
         } else {
-            // Create frame and set structural metadata
-            size_t num_vis = _stack.size() > 0 ? _stack.size() : _prods.size();
-
-            auto frame = VisFrameView::create_frame_view(out_buf, frame_id, _inputs.size(), num_vis,
-                                                         _ev.size());
-
-            // Fill data with zeros
-            std::fill(frame.vis.begin(), frame.vis.end(), 0.0);
-            std::fill(frame.weight.begin(), frame.weight.end(), 0.0);
-            std::fill(frame.eval.begin(), frame.eval.end(), 0.0);
-            std::fill(frame.evec.begin(), frame.evec.end(), 0.0);
-            std::fill(frame.gain.begin(), frame.gain.end(), 0.0);
-            std::fill(frame.flags.begin(), frame.flags.end(), 0.0);
-            frame.erms = 0;
-
-            // Set non-structural metadata
-            frame.freq_id = 0;
-            frame.dataset_id = dset_id_t::null;
-            frame.time = std::make_tuple(0, timespec{0, 0});
-
-            // mark frame as empty by ensuring this is 0
-            frame.fpga_seq_length = 0;
-            frame.fpga_seq_total = 0;
-
-            DEBUG("visRawReader: Reading empty frame: {:d}", frame_id);
+            // Create empty frame and set structural metadata
+            create_empty_frame(frame_id);
         }
 
         // Set the dataset ID to the updated value
-        dset_id_t& ds_id = ((VisMetadata*)(out_buf->metadata[frame_id]->metadata))->dataset_id;
+        dset_id_t& ds_id = get_dataset_id(frame_id);
         ds_id = get_dataset_state(ds_id);
```

Comment:
The naming of these two functions and the way they are used makes sense individually, but in combination is somewhat confusing, i.e it implies you get a dataset_id and then assign a dataset_state to it, despite those being different types and concepts.